### PR TITLE
Make cp be able to copy files to directory

### PR DESCRIPTION
**Pull request by alexxbb**
This PR makes the cp command be able to copy files to a directory, similar to what bash supports: 
This doesn't yield huge productivity, but it saves a few keystrokes and makes the behaviour closer to that of a shell. 
It also temporary "fixes" a test which was failing on MacOs since it doesn't support `date --iso`.
```rust
// currently 
let src = "/some/file.txt";
cp(src, Path::new("./directory").join(src));
// with this PR
cp(src, "./directory");
```

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -41,8 +41,12 @@
...
 //! ```
 //! # use xshell::cmd;
-//! let output = cmd!("date --iso").read()?;
-//! assert!(output.chars().all(|c| "01234567890-".contains(c)));
+//! if cfg!(target_os = "macos") {
+//!     let output = cmd!("date").read()?;
+//! } else {
+//!     let output = cmd!("date --iso").read()?;
+//!     assert!(output.chars().all(|c| "01234567890-".contains(c)));
+//! }
```
Comment (matklad) [inline src/lib.rs:49]: Let's revert this change for now.

I haven't put time into fixing the tests for other OSes yet, but the way to do this would be to implement dummy `date` and `echo` ourselves, as binaries in `example`.

File: src/fs.rs [hunk 2]
```diff
@@ -42,6 +42,14 @@ pub fn cp(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> Result<()> {
 }
 fn _cp(src: &Path, dst: &Path) -> Result<()> {
     let _guard = gsl::read();
+    let mut dst = dst;
```
Comment (matklad) [inline src/fs.rs:45]:
```suggestion
fn _cp(src: &Path, mut dst: &Path) -> Result<()> {
    let _guard = gsl::read();
```

File: src/fs.rs [hunk 3]
```diff
@@ -42,6 +42,14 @@ pub fn cp(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> Result<()> {
...
+    let mut _tmp;
+    if dst.is_dir() {
+        if let Some(trunk) = src.file_name() {
+            _tmp = dst.join(trunk);
```
Comment (matklad) [inline src/fs.rs:49]:
```suggestion
        if let Some(file_name) = src.file_name() {
            _tmp = dst.join(file_name);
```

**Review by matklad — COMMENTED** [review]
I like this! 

Long term (definitelly not in this PR), I'd love to see `cp_r` as well, theres sadly no intutitive file copying crate in the ecosystem at the moment.

**Comment by matklad** [comment]
Added xplatform CI in https://github.com/matklad/xshell/pull/6

File: src/fs.rs [hunk 4]
```diff
@@ -42,11 +42,11 @@ pub fn cp(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> Result<()> {
 }
 fn _cp(src: &Path, dst: &Path) -> Result<()> {
     let _guard = gsl::read();
-    let mut dst = dst;
     let mut _tmp;
+    let mut dst = dst;
```
Comment (alexxbb) [inline src/fs.rs:46]: Making `mut dst: &Path` didn't work, because of the lifetime issue where _tmp didn't live long enough.
